### ISSUE TITLE
Move to a per route POIs data model

### ISSUE THREAD

**Issue by HarelM** [IsraelHikingMap/Site#272]
I think that every route should have its POIs and not a POI bucket for all the routes.
Proposed changes:
1. Save button will only save the current route and its POIs.
2. Save as will facilitate for "Save all routes to a single GPX file" to let the user know that some data is begin lost, same as Single GPX track.
3. Hiding a route will hide its POIs too.
4. Loading files will not collide.
5. There won't be a save button inside the route properties dialog.
6. Undo and clear buttons will work on both POIs and route at the same time.

There is an issue with search results when it creates multiple routes, I'm not sure how to solve this UX wise (maybe to open the layers menu so that the user will know more than one route was created).

Share will continue to work as it does today.
Database will be migrated but no data will be lost.

@zstadler let me know if I missed something, I know we discussed a two types of layer model, but I think it will only complicate the UX...

**Comment by zstadler** [comment]
@HarelM there were at least two rounds of discussions on this issue in different times, but now I've lost memory of the different approaches and the pros and cons of each. Can you summarize them?

Let's double-check the alternatives before discussing the specific implementation.

**Comment by HarelM** [comment]
Not sure I understand your question, but let's see if I can elaborate:
Pros:
When opening multiple files the POI data won't get mixed.
You'll be able to manage per route POIs.
Cons:
When having multiple routes you'll need to know how to save all of then to a file which won't be straight forward.

I think that the pros outweigh the cons.

We then moved to discuss how to represent it in the layers control.
You suggested two types of layers while I suggested a single type of later that can hold both POI and route data.

**Comment by zstadler** [comment]
Thanks for the refresh!

Indeed the only drawback we've found to this approach is that the association between waypoints and tracks is lost when saving to a GPX/TWL/CSV file and reloading it. It is not an issue with creating a share in the site. KML files can also be written in a way that will preserve this association.

I agree the pros outweigh the cons, especially wen loading multiple overlapping traces from different sources, each with its set of POIs.

Technically, we can use GPX extensions to maintain the association of POIs to tracks, but this is a low priority.
